### ISSUE TITLE
bug: Duplicate of the same enchant when using enchantment table

### ISSUE THREAD

**Issue by Wide0125** [adibarra/enchant-tweaker#20]
### What happened?

There is a not too rare chance to get duplicates of the same enchants when using the enchantment table to enchant items. After testing, found it applied to swords and armor pieces.

### What should have happened?

Only one of that enchantment should've been applied

### How to reproduce?

Use an enchantment table to enchant armor or swords at level 30 a couple times.

### What is the jar file name?

enchanttweaker-1.4.5+mc1.20-1.20.1.jar

### What Minecraft version?

Fabric 1.20.2

### Any relevant logs?

https://mclo.gs/UOn2Ze8

### Anything else?

![Screenshot 2023-12-19 at 10 22 43 PM](https://github.com/adibarra/enchant-tweaker/assets/95921629/50c66a00-e73a-4f43-b73f-ee6373fc318d)
![Screenshot 2023-12-19 at 10 22 41 PM](https://github.com/adibarra/enchant-tweaker/assets/95921629/b17f54f1-5850-4c5e-bb38-e6042f486d52)
![Screenshot 2023-12-19 at 10 22 39 PM](https://github.com/adibarra/enchant-tweaker/assets/95921629/df766600-e368-43e6-a72f-6152e2cca8ae)
![Screenshot 2023-12-19 at 10 19 38 PM](https://github.com/adibarra/enchant-tweaker/assets/95921629/127d9537-de03-4ba9-b06b-beedd6a4f8eb)

### Before submitting...

- [X] This issue is not a duplicate of an existing issue (open or closed).

**Comment by adibarra** [comment]
Hello, thanks for the report! I'll start looking into this. Just to clarify were you using any other mods at the same time? I'd appreciate it if you could attach a logfile (and/or mod list) if you have one.

**Comment by Wide0125** [comment]
![Screenshot 2023-12-22 at 12 24 02 AM](https://github.com/adibarra/enchant-tweaker/assets/95921629/f95a0aca-8b86-424b-abc5-faa5c4f259cb)
I've added our villager discounts, rcsd, and enchanting tweaks (to implement the god armor and god sword functionality with no bug) since I encountered the bug.

**Comment by JairusSW** [comment]
I also am getting the same issue.
![image](https://github.com/adibarra/enchant-tweaker/assets/60235514/4f955666-11e2-4bff-b13d-27dc59e31f40)

**Comment by adibarra** [comment]
I was able to reproduce it as well and have been trying to track it down. I just haven't had the time to nail it down yet. Sorry about the wait on this one.

**Comment by JairusSW** [comment]
Hey, don't worry. It's all good

**Comment by adibarra** [comment]
It should be fixed now. The fix will be included in the next update (v1.4.7)

**Comment by JairusSW** [comment]
Confirmed. Working in latest release. Thank you!
